docs/rfc/active/000-internationalization.md

```diff
@@ -0,0 +1,165 @@
+> - [ ] Factorisation of Dates
+> - [ ] Factorisation of numbers (count, filter, table total, etc)
+
+### Extensibility
+> Anybody can add his own language by adding the a *translation.json* file in the corresponding directory.
+
+## Non-Requirements
+> I won't talk about which language should be implemented or not. English is native. 
+> I will take care of French. If anyone want his own language, let him be. 
+
+## Detailed design
+
+### Terminology
+> i18n : Internationalisation (begin with an "i", end with a "n" and have 18 letters between)
+>
+
+### Overview
+> I am using [react-i18next](https://react.i18next.com/) to do so. Basically we wrapp the App in a I18nestProvider
+> which allow us to call a hook or a render prop. We give them a key string which point on a *translation.json*
+> containing the text we want (according to the choosen language). Example from the react-i18next doc:
+```javascript
+// Example hook
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+
+export function MyComponent() {
+    const { t, i18n } = useTranslation();
+    // or const [t, i18n] = useTranslation();
+
+    return <p>{t('type.translatedText')}</p>
+}
+```
+```javascript
+// Example component
+import React from 'react';
+import { Trans } from 'react-i18next';
+
+export function MyComponent() {
+    return (
+        <Trans i18nKey="type.translatedText"/>
+    )
+}
+```
+> You can also pass dynamic text, or pass some html (but you have to use the Trans component for that).
+> 
+
+> In any case, if the translation doesn't exist in requested language, then it give the default (ie. English) translation.
+> If it doesn't exist in the default language, then it return the given translation key.
+> So the worste case scenario is an ugly translation, but no blocking bugs.
+>
+
+> By default, it can be configured to use the browser language
+> (with a fallback to the default language if browser language is not translated).
+>
+
+> In practice, I think the lightest way for the Datahub team would be to just implement the english language. 
+> Leaving any other language for others devs. Since we should be able to dynamically check how many language there is and which one are they,
+> we could add a "Change Language" field in the Header Menu opening a modal where you can select your desired language.
+> And obviously, we hide it if there is only one language selectable.
+
+
+### Corner-case
+
+#### Front end test
+> There are tests in the frontend looking for the presence of text. I have deactivated them.
+> I am not sure how react-i18next will render in a test environment.
+> 
+
+#### Back end text
+> Some text come from the back end.
+> - The back-end exceptions : we can handle then by adding a translate key to exceptions, and the front will
+    > have to apply i18next on the the translate key to display it.
+> - The whole Analytics page : It is one of the majors problem... All text of the Analytics page is built,
```

Comment:
Maybe a backend translation file as well..

Reply:
That's a possibility... I'll have to check the existing i18n libraries for Java.